src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingSpec.cs

```diff
@@ -219,20 +220,36 @@ protected override void PostStop()
 
         protected override bool ReceiveRecover(object message)
         {
-            return message.Match()
-                .With<CounterChanged>(UpdateState)
-                .WasHandled;
+            switch (message)
+            {
+                case CounterChanged cc:
+                    UpdateState(cc);
+                    return true;
+            }
+            return false;
```

Comment:
@Horusiath Is this a valid replacement for the message.Match().With<> semantics ?

Reply:
Yes

Reply:
ok

Reply:
why the switch statement on only one case?  it would be better expressed as an if statement.